homeassistant/components/device_tracker/unifi_direct.py

```diff
@@ -0,0 +1,134 @@
+"""
+Support for Unifi AP direct access.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.unifi_direct/
+"""
+import logging
+import json
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME,
+    CONF_PORT)
+
+REQUIREMENTS = ['pexpect==4.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SSH_PORT = 22
+UNIFI_COMMAND = 'mca-dump | tr -d "\n"'
+UNIFI_SSID_TABLE = "vap_table"
+UNIFI_CLIENT_TABLE = "sta_table"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_SSH_PORT): cv.port
+})
+
+
+# pylint: disable=unused-argument
+def get_scanner(hass, config):
+    """Validate the configuration and return a Unifi direct scanner."""
+    try:
+        return UnifiDeviceScanner(config[DOMAIN], hass)
+    except (ConnectionError, KeyError):
+        return False
+
+
+class UnifiDeviceScanner(DeviceScanner):
+    """This class queries Unifi wireless access point."""
+
+    def __init__(self, config, hass):
+        """Initialize the scanner."""
+        self.host = config[CONF_HOST]
+        self.username = config[CONF_USERNAME]
+        self.password = config[CONF_PASSWORD]
+        self.port = config[CONF_PORT]
+        self.ssh = None
+        self.connected = False
+        self.last_results = {}
+
+    def scan_devices(self):
+        """Scan for new devices and return a list with found device IDs."""
+        result = _response_to_json(self._get_update())
+        if result is not None:
+            self.last_results = result
+        return self.last_results.keys()
+
+    def get_device_name(self, device):
+        """Return the name of the given device or None if we don't know."""
+        hostname = next((
+            value.get('hostname') for key, value in self.last_results.items()
+            if key.upper() == device.upper()), None)
+        if hostname is not None:
+            hostname = str(hostname)
+        return hostname
+
+    def _connect(self):
+        """Connect to the Unifi AP SSH server."""
+        from pexpect import pxssh
+
+        self.ssh = pxssh.pxssh()
+        self.ssh.login(self.host, self.username,
+                       password=self.password, port=self.port)
+
+        self.connected = True
+
+    def _disconnect(self):
+        """Disconnect the current SSH connection."""
+        # pylint: disable=broad-except
+        try:
+            self.ssh.logout()
+        except Exception:
+            pass
```

Comment:
I would advice against this. Broad except and no action if the exception occurs is usually a bad combo. Preferably catch a specific error.

Reply:
So the problem is, I don't know what happens. I also don't really care what happens, but if something bad happens I really want the finally block to run so the connection is set to None.